src/librustc_resolve/lib.rs

```diff
@@ -1407,6 +1408,72 @@ impl<'a, 'b: 'a> ty::DefIdTree for &'a Resolver<'b> {
 
 impl<'a> hir::lowering::Resolver for Resolver<'a> {
     fn resolve_hir_path(&mut self, path: &mut hir::Path, is_value: bool) {
+        self.resolve_hir_path_cb(path, is_value,
+                                 |resolver, span, error| resolve_error(resolver, span, error))
+    }
+
+    fn resolve_str_path(&mut self, span: Span, crate_root: Option<&str>,
+                components: &[&str], is_value: bool) -> hir::Path {
+        use std::iter;
```

Comment:
I don't see any reason why it couldn't be imported at the top of the file. Maybe I missed something?

Reply:
For imports you'll probably only need in one place it's common practice to do this.